packages/@sanity/base/src/search/weighted/createWeightedSearch.ts

```diff
@@ -2,85 +2,60 @@
 import type {SanityClient} from '@sanity/client'
 import type {ObjectSchemaType} from '@sanity/types'
 import type {Observable} from 'rxjs'
-import {compact, toLower, flatten, uniq, flow, sortBy, union} from 'lodash'
+import sortBy from 'lodash/sortBy'
 import {map, tap} from 'rxjs/operators'
-import {joinPath} from '../../util/searchUtils'
-import {tokenize} from '../common/tokenize'
 import {removeDupes} from '../../util/draftUtils'
 import {applyWeights} from './applyWeights'
-import {WeightedHit, WeightedSearchOptions, SearchOptions, SearchPath, SearchHit} from './types'
+import {SearchHit, SearchOptions, SearchTerms, WeightedHit, WeightedSearchOptions} from './types'
+import {createSearchQuery} from './createSearchQuery'
 
-const combinePaths = flow([flatten, union, compact])
-
-const toGroqParams = (terms: string[]): Record<string, string> => {
-  const params: Record<string, string> = {}
-  return terms.reduce((acc, term, i) => {
-    acc[`t${i}`] = `${term}*` // "t" is short for term
-    return acc
-  }, params)
+type ObjectSchema = {
+  name: string
+  // eslint-disable-next-line camelcase
+  __experimental_search: ObjectSchemaType['__experimental_search']
 }
```

Comment:
This is duplicated both here and in `createWeightedSearch.ts` - and the naming is quite similar to `ObjectSchemaType` from `@sanity/types`. I'd suggest renaming it to `SearchableType` (or similar), and share it between the functions that accepts it).